### DISCUSSION TITLE
MVT Layer - getRenderedFeatures

### DISCUSSION THREAD

**Discussion by JG145** [visgl/deck.gl#5698] [Q&A]
For the life of me I cannot get the method - getRenderedFeatures to work using DeckGL.

Could someone please show a working example of the code in use?

thanks

**Comment by AdriSolid** [comment]
You will need to do some things:

- Access `getRenderedFeatures` method from the layer instance.
- Requires `uniqueIdProperty`
- Requires `pickable` = `true`
- Wait for `onViewStateChange` with some debounce.

[Here an example.](https://codepen.io/AdriSolid/pen/QWdYYoe?editors=1010)

Hope it helps!

**Reply by JG145** [reply]
Thanks - got it partially working - but it is only returning a single value and not the full array - any ideas?

**Reply by AdriSolid** [reply]
Try to reproduce your issue in a codepen please.
